backend/hdf5/DimensionHDF5.cpp

```diff
@@ -259,13 +269,153 @@ void SetDimensionHDF5::labels(const none_t t) {
 
 SetDimensionHDF5::~SetDimensionHDF5() {}
 
+//--------------------------------------------------------------
+// Implementation of DataFrameDimensionHDF5
+//--------------------------------------------------------------
+DataFrameDimensionHDF5::DataFrameDimensionHDF5(const H5Group &group, ndsize_t index)
+    : DimensionHDF5(group, index)
+{
+    setType();
+    // Note: maybe having this is a bad idea, because a DataFrameDimension created
+    // this way will not have the block and file
+}
+
+DataFrameDimensionHDF5::DataFrameDimensionHDF5(const H5Group &group, ndsize_t index,
+                                               const std::shared_ptr<IFile> &file,
+                                               const std::shared_ptr<IBlock> &block)
+    : DimensionHDF5(group, index), entity_block(block), entity_file(file)
+{
+    setType();
+}
+
+DataFrameDimensionHDF5::DataFrameDimensionHDF5(const H5Group &group, ndsize_t index,
+                                               const std::shared_ptr<IFile> &file,
+                                               const std::shared_ptr<IBlock> &block, const DataFrame &frame)
+    :DataFrameDimensionHDF5(group, index, file, block)
+{
+    std::shared_ptr<IDataFrame> idf = block->getEntity<IDataFrame>(frame.id());
+    if (!idf)
+        throw std::runtime_error("ColumnDimensionHDF5 DataFrame not found in block!");
+    if (this->group.hasGroup("data_frame"))
+        this->group.removeGroup("data_frame");
+    auto target = std::dynamic_pointer_cast<DataFrameHDF5>(idf);
+
+    setType();
+    this->group.createLink(target->group(), "data_frame");
+}
+
+DataFrameDimensionHDF5::DataFrameDimensionHDF5(const H5Group &group, ndsize_t index,
+                                               const std::shared_ptr<IFile> &file,
+                                               const std::shared_ptr<IBlock> &block, const DataFrame &frame,
+                                               unsigned col_index)
+    :DataFrameDimensionHDF5(group, index, file, block, frame)
+{
+    this->group.setAttr("column_index", col_index);
+}
+
+int DataFrameDimensionHDF5::columnIndex() const {
+    int col_index = -1;
+    if (group.hasAttr("column_index")) {
+        group.getAttr("column_index", col_index);
+    }
+    return col_index;
+}
+
+int DataFrameDimensionHDF5::checkColumnIndex(int col_index) const {
+    int column_index = col_index;
+    if (col_index == -1) {
+        column_index = columnIndex();
+    }
+
+    if (column_index == -1) {
+        throw nix::OutOfBounds("DataFrameDimension: Error accessing column, no column index was given and no default is specified.");
```

Comment:
Sometimes the error have `HDF5` suffix, sometimes not.

Reply:
done, all get suffixes